### ISSUE TITLE
Package Mmap [44cfe95a-1eb2-52ea-b672-e2afdf69b78f] is required but does not seem to be installed

### ISSUE THREAD

**Issue by staticfloat** [JuliaLang/Pkg.jl#1015]
I just created a new environment, added a bunch of packages, then at `]build` time, I reliably get this:

```
┌ Error: Error building `PyCall`:
│ ERROR: LoadError: LoadError: ArgumentError: Package Mmap [44cfe95a-1eb2-52ea-b672-e2afdf69b78f] is required but does not seem to be installed:
│  - Run `Pkg.instantiate()` to install all recorded dependencies.
│
│ Stacktrace:
│  [1] _require(::Base.PkgId) at ./loading.jl:929
│  [2] require(::Base.PkgId) at ./loading.jl:858
│  [3] macro expansion at ./logging.jl:311 [inlined]
│  [4] require(::Module, ::Symbol) at ./loading.jl:840
│  [5] include at ./boot.jl:317 [inlined]
│  [6] include_relative(::Module, ::String) at ./loading.jl:1038
│  [7] include at ./sysimg.jl:29 [inlined]
│  [8] include(::String) at /home/sabae/.julia/packages/JSON/ebvl3/src/JSON.jl:3
│  [9] top-level scope at none:0
│  [10] include at ./boot.jl:317 [inlined]
│  [11] include_relative(::Module, ::String) at ./loading.jl:1038
│  [12] include(::Module, ::String) at ./sysimg.jl:29
│  [13] top-level scope at none:2
│  [14] eval at ./boot.jl:319 [inlined]
│  [15] eval(::Expr) at ./client.jl:396
│  [16] top-level scope at ./none:3
│ in expression starting at /home/sabae/.julia/packages/JSON/ebvl3/src/Parser.jl:3
│ in expression starting at /home/sabae/.julia/packages/JSON/ebvl3/src/JSON.jl:11
│ ERROR: LoadError: Failed to precompile JSON [682c06a0-de6a-54ab-a142-c8b1cf79cde6] to /home/sabae/.julia/compiled/v1.1/JSON/uf6oy.ji.
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] macro expansion at ./logging.jl:313 [inlined]
│  [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
│  [4] _require(::Base.PkgId) at ./logging.jl:311
│  [5] require(::Base.PkgId) at ./loading.jl:858
│  [6] macro expansion at ./logging.jl:311 [inlined]
│  [7] require(::Module, ::Symbol) at ./loading.jl:840
│  [8] include at ./boot.jl:317 [inlined]
│  [9] include_relative(::Module, ::String) at ./loading.jl:1038
│  [10] include(::Module, ::String) at ./sysimg.jl:29
│  [11] top-level scope at none:2
│  [12] eval at ./boot.jl:319 [inlined]
│  [13] eval(::Expr) at ./client.jl:396
│  [14] top-level scope at ./none:3
│ in expression starting at /home/sabae/.julia/packages/Conda/uQitS/src/Conda.jl:19
│ ERROR: LoadError: Failed to precompile Conda [8f4d0f93-b110-5947-807f-2305c1781a2d] to /home/sabae/.julia/compiled/v1.1/Conda/WZE3U.ji.
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] macro expansion at ./logging.jl:313 [inlined]
│  [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
│  [4] _require(::Base.PkgId) at ./logging.jl:311
│  [5] require(::Base.PkgId) at ./loading.jl:858
│  [6] macro expansion at ./logging.jl:311 [inlined]
│  [7] require(::Module, ::Symbol) at ./loading.jl:840
│  [8] include at ./boot.jl:317 [inlined]
│  [9] include_relative(::Module, ::String) at ./loading.jl:1038
│  [10] include(::Module, ::String) at ./sysimg.jl:29
│  [11] include(::String) at ./client.jl:395
│  [12] top-level scope at none:0
│ in expression starting at /home/sabae/.julia/packages/PyCall/0jMpb/deps/build.jl:9
└ @ Pkg.Operations ~/src/julia/usr/share/julia/stdlib/v1.1/Pkg/src/Operations.jl:1069
```

It appears to me that since `JSON` does not have a `Project.toml` file, it should not need to set an explicit dependency on an stdlib, but this is a different error than I am used to, as usually Pkg just complains that `Mmap` is not in the `Project.toml` and loads it anyway.  It's almost like `Pkg` isn't recognizing `Mmap` as an stdlib.

This is using Julia from the `kf/tpu3` branch:

```
julia> versioninfo()
Julia Version 1.1.0-DEV.581
Commit f1dffc5c8b* (2019-01-21 07:10 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU @ 2.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, ivybridge)
```

**Comment by 00vareladavid** [comment]
I'm confused, I thought `JSON` has no build step.

**Comment by staticfloat** [comment]
I have attempted to explicitly add `Mmap` to my project, to no avail.  Here are the [`Manifest.toml` and `Project.toml` files](https://gist.github.com/staticfloat/6b7c7b42305ddcff99954d50624d3f48), in case anyone wants to take a look.

**Comment by staticfloat** [comment]
Hmm, I have no idea where that UUID for `Mmap` is coming from (`44cfe95a-1eb2-52ea-b672-e2afdf69b78f`).  Every `Manifest.toml` file I open has the UUID `a63ad114-7e13-5084-954f-fe012c677804` listed for it.

**Comment by KristofferC** [comment]
`44cfe95a-1eb2-52ea-b672-e2afdf69b78f` is the UUID for Pkg...

**Comment by KristofferC** [comment]
What if you do `Pkg.Types.Context().stdlibs`? What does it show?

**Comment by staticfloat** [comment]
```
julia> import Pkg

julia> Pkg.Types.Context().stdlibs
Dict{Base.UUID,String} with 29 entries:
  UUID("4607b0f0-06f3-5cda-b6b1-a6196a1729e9") => "SuiteSparse"
  UUID("2a0f44e3-6c83-55bd-87e4-b1978d98bd5f") => "Base64"
  UUID("8ba89e20-285c-5b6f-9357-94700520ee1b") => "Distributed"
  UUID("56ddb016-857b-54e1-b83d-db4d58db5568") => "Logging"
  UUID("a63ad114-7e13-5084-954f-fe012c677804") => "Mmap"
  UUID("ade2ca70-3891-5945-98fb-dc099432e06a") => "Dates"
  UUID("de0858da-6303-5e67-8744-51eddeeeb8d7") => "Printf"
  UUID("ea8e919c-243c-51af-8825-aaa63cd721ce") => "SHA"
  UUID("6462fe0b-24de-5631-8697-dd941f90decc") => "Sockets"
  UUID("8dfed614-e22c-5e08-85e1-65c5234f0b40") => "Test"
  UUID("4ec0a83e-493e-50e2-b9ac-8f72acf5a8f5") => "Unicode"
  UUID("b77e0a4c-d291-57a0-90e8-8db25a27a240") => "InteractiveUtils"
  UUID("8f399da3-3557-5675-b5ff-fb832c97cbdb") => "Libdl"
  UUID("9a3f8284-a2c9-5f02-9a11-845980a1fd5c") => "Random"
  UUID("9e88b42a-f829-5b0c-bbe9-9e923198166b") => "Serialization"
  UUID("1a1011a3-84de-559e-8e89-a11a2f7dc383") => "SharedArrays"
  UUID("10745b16-79ce-11e8-11f9-7d13ad32a3b2") => "Statistics"
  UUID("3fa0cd96-eef1-5676-8a61-b3b8758bbffb") => "REPL"
  UUID("7b1f6079-737a-58dc-b8bc-7a2ca5c1b5ee") => "FileWatching"
  UUID("9abbd945-dff8-562f-b5e8-e1ebf5ef1b79") => "Profile"
  UUID("44cfe95a-1eb2-52ea-b672-e2afdf69b78f") => "Pkg-cfbe0479e609aabcf5ae5422f7772b742922a0da"
  UUID("d6f4376e-aef5-505a-96c1-9c027394607a") => "Markdown"
  UUID("8bb1440f-4735-579b-a4ab-409b98df4dab") => "DelimitedFiles"
  UUID("76f85450-5226-5b5a-8eaa-529ad045b433") => "LibGit2"
  UUID("cf7118a7-6976-5b1a-9a39-7adc72f591a4") => "UUIDs"
  UUID("8bf52ea8-c179-5cab-976a-9e18b702a9bc") => "CRC32c"
  UUID("37e2e46d-f89d-539d-b4ee-838fcccc9c8e") => "LinearAlgebra"
  UUID("2f01184e-e22b-5df5-ae63-d93ebab69eaf") => "SparseArrays"
  UUID("9fa8497b-333b-5362-9e8d-4d0656e87820") => "Future"
```

**Comment by KristofferC** [comment]
>   UUID("44cfe95a-1eb2-52ea-b672-e2afdf69b78f") => "Pkg-cfbe0479e609aabcf5ae5422f7772b742922a0da"

This dude looks a bit suspicious...

Could you go into `julia/stdlib` and delete any weird `Pkg` folder (and perhaps rerun `make` to get a normal `Pkg` in there).

**Comment by staticfloat** [comment]
Ah, hmmm it looks like I have two `Pkg`'s in there:

```
$ ll stdlib/ | grep Pkg
drwxr-xr-x  7 sabae sabae 4096 Dec 13 23:26 Pkg/
drwxr-xr-x  7 sabae sabae 4096 Dec 13 22:57 Pkg-cfbe0479e609aabcf5ae5422f7772b742922a0da/
```

Removing them and re-running `make` caused an error, because apparently `stdlib/Pkg` is actually checked into the repository.  It looks like this version of Julia predates the move of `Pkg` to a vendored package.  I did a `git checkout stdlib/Pkg`, which restored the files, then re-ran `make` which built successfully, and finally re-ran `]build` which worked like a charm.  I'm betting what happened was I built a newer version of Julia in this directory, then switched to an older one, and that mysterious `Pkg-cfbe04...` directory somehow interfered with the sysimg build.

Thanks!
